cinder/volume/api.py

```diff
@@ -799,9 +806,40 @@ def initialize_connection(self, context, volume, connector):
             msg = _("The volume connection cannot be initialized in "
                     "maintenance mode.")
             raise exception.InvalidVolume(reason=msg)
-        init_results = self.volume_rpcapi.initialize_connection(context,
-                                                                volume,
-                                                                connector)
+
+        def _migrate_and_initialize_connection():
+            volume_type = {}
+            if volume.volume_type_id:
+                volume_type = volume_types.get_volume_type(
+                    context.elevated(), volume.volume_type_id)
+            request_spec = {
+                'volume_properties': volume,
+                'volume_type': volume_type,
+                'volume_id': volume.id}
+            try:
+                dest = self.scheduler_rpcapi.find_backend_for_connector(
+                    context, connector, request_spec)
+            except exception.NoValidBackend:
+                LOG.exception("The current backend couldn't be used with"
+                              "the provided connector and couldn't find"
+                              "another backend to migrate the volume to.")
```

Comment:
Since this is inside of the initialize_connection api call, it might be confusing and odd to see a failure to find a backend to migrate to.   Can we be a little more descriptive here?